### PR TITLE
grade-school: add canonical data

### LINKED ISSUE #561: grade-school: Implement canonical-data.json
We want to have a standard set of test inputs and outputs for each exercise to
make it easier to port them to new languages, as well as to help keep the
exercises in sync.

The Grade School exercise can be found in the [./exercises/grade-school/][exercise-dir] directory.

The step-by-step instructions for how to compile a canonical-data.json file is
described in [this section of the contributing guide][contributing].

See http://exercism.io/contribute/canonical-data/grade-school for the up-to-date list of
language tracks that have an implementation of the Grade School exercise.

The [tracking issue][tracking-issue] has more context.

[contributing]: https://github.com/exercism/docs/blob/master/you-can-help/improve-exercise-metadata.md#extracting-canonical-test-data
[tracking-issue]: https://github.com/exercism/x-common/issues/552
[exercise-dir]: https://github.com/exercism/x-common/tree/master/exercises/grade-school

### PR DESCRIPTION
**Pull request by rpottsoh**
closes #561
[description](https://github.com/exercism/problem-specifications/blob/master/exercises/grade-school/description.md)

This testdata is based on implementations of this exercise in the csharp and Delphi tracks.

### CODE REVIEW

File: exercises/grade-school/canonical-data.json [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+            "property": "grade",
+            "input": {
+              "students": [],
+              "desiredGrade": 1
+            },
+            "expected": []
+        },
+        {
+            "description": "Roster returns an empty list if there are no students enrolled",
+            "property": "roster",
```
[the remark this answers is not in the document]
Comment (rpottsoh) [reply exercises/grade-school/canonical-data.json]: No specific reason.  I added `Roster returns an empty list if there are no students enrolled` when I implemented the exercise in Delphi.  The rest of the tests are in the same order as they appear in csharp.  I have no qualms with changing the order of the tests.

File: exercises/grade-school/canonical-data.json [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    "comments": [
+        "Given students' names along with the grade that they are in, ",
+        "create a roster for the school."
+    ],
+    "cases": [
+        {
+            "description": "Adding a student adds them to the sorted roster",
+            "property": "roster",
+            "input": {
+              "students": [["Aimee",2]]
```
[the remark this answers is not in the document]
Comment (rpottsoh) [reply exercises/grade-school/canonical-data.json]: > In general, I believe array items are separated with , (with a space).
This also goes for the rest of the document.

Done,

File: exercises/grade-school/canonical-data.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (ErikSchierboom) [reply exercises/grade-school/canonical-data.json]: I think I would indeed prefer a re-ordering. Other than that, it looks great!

**Comment by rpottsoh** [comment]
If there are no other changes to be made or suggestions to be heard then I believe this PR is ready to be merged.

**Pull request merged**